### DISCUSSION TITLE
Add ReplyTo  to Send Email operation

### DISCUSSION THREAD

**Discussion by Pachat** [directus/directus#19989] [Feature Requests]
### Summary

it would be very useful for cloud users to at a minimum specify a replyTo since we cannot change the SMTP and sender.
(changing the sender email would understandably be a bridge too far and affect the send rate without SPF and DKIM.)

### Motivation

Sending mails from different users without Reply-To sends all responses to the SMTP user email that has the to dispatch them.

### Detailed Design

Add a field Reply-To in the mail operation form

### Must Have:
- A
- B

### Should Have:
- C

### Could Have:
- D

### Won't Have:
- E

### Drawbacks

-

### Alternatives

-

### Adoption Strategy

-

**Comment by Pachat** [comment]
Looks like https://github.com/directus/directus/discussions/16726#discussioncomment-4324604 is giving a solution

**Reply by yannbolliger** [reply]
I would still very much like to see this implemented in Directus itself. Would also be willing to provide a PR, maybe with some direction?

**Reply by yannbolliger** [reply]
@rijkvanzanten would it be enough to add the params `replyTo, cc, bcc` [here](https://github.com/directus/directus/blob/main/api/src/operations/mail/index.ts#L21)? Because as far as I see, they'll get passed to nodemailer in the MailService. Then, they could be added as fields in the [app/UI](https://github.com/directus/directus/blob/main/app/src/operations/mail/index.ts) definition...

**Reply by rijkvanzanten** [reply]
My main concern is spam protection. Seeing we offer the Directus users to now send emails under a different replyTo email, it opens up the smtp server to reputation problems. Email deliverability is a notoriously finicky system, so we have to be very careful with what we support within emails. Maybe we can find a way where we can unlock some of those functionalities when the end user recognizes those risks on their sending infrastructure through an env var or something similar, but that needs a bit more thought

**Reply by yannbolliger** [reply]
I understand the concern. However, for our use-case setting a reply-to is crucial: We have a contact form for our (~30) team members which is based on a Directus flow. Currently, our members get emails from a `no-reply@our-domain.com`. This is a problem because in most cases, they want to reply to the person that filled out the contact form, they hit reply and the mail goes nowhere. For us, it would therefore be important to be able to set the form fillers email as reply-to.

**Comment by Pachat** [comment]
@rijkvanzanten 
>the end user recognizes those risks on their sending infrastructure through an env var or something similar

I totally agree with the emphasis on the user responsibility. 
One could restrict reply-to to emails of already registered users.

Or, more strictly, the user should check a box on their profile to indicate that emails to be sent with a reply-to that's their profile email address.

Or, even more simply, this could be a global parameter in the .env
SMTP_REPLY_TO_USER_EMAIL=true

**Comment by pztrick** [comment]
I would also like to be able to customize the Sender name (and possibly email, currently set by `EMAIL_FROM` to e.g. `no-reply@example.com`) for Flows which send email.

It could make sense to just add a database model/table around Sender Identities, and include columns for marking whether the Sender is verified to provide visibility around email deliverability. For example, in an Email Flow you would have a drop-down menu for the Sender Identity from a list populated from the database table... When you go into the Sender Identities page you can create/delete senders and see checkmarks for whether the domain name or sender is verified. At least for the Sendgrid provider, I know you can query the API using the api key for verified domains and verified single email addresses... These could populate columns in the Senders settings page to indicate deliverability should be good.

Sendgrid docs: https://www.twilio.com/docs/sendgrid/for-developers/sending-email/sender-identity

For other email services, we could also just have Senders created in the "inactive" state which before using you would have to view some modal window indicating that you should send a test email to a service like `mail-tester.com` to evaluate deliverability and make any DKIM/SPF/whatever changes. If you click "Confirm" on the modal you accept responsibility for any email deliverability configuration on your end.

**Comment by Pachat** [comment]
Just to let you know that it was that crucial for us, AFAIR we hacked the code (old version v10.2.1) 
We send the mail from `no-replye@..` with `Reply-To` set to the user who sends the email with no problem since half a year.

**Reply by yannbolliger** [reply]
@Pachat would you mind publishing the code changes you did in a gist or fork?
